### PR TITLE
apktool: 2.8.0 -> 2.8.1 and use free aapt

### PR DESCRIPTION
**Pull request by linsui**
###### Description of changes

The android sdk is licensed under an unfree license but the aapt from Google Maven is [licensed under Apache-2.0 only](https://maven.google.com/web/index.html#com.android.tools.build:aapt2:8.0.2-9289358). So I extract it there for a free aapt package. This also decrease the closure size of apktool from 2.2G to 599.5M.

Since apktool is free now, the unfree flag in diffoscope is not needed anymore.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/aapt/default.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  src =
+    let
+      urlAndHash =
+        if stdenvNoCC.isLinux then {
+          url = "https://dl.google.com/android/maven2/com/android/tools/build/aapt2/${version}/aapt2-${version}-linux.jar";
+          hash = "sha256-P8eVIS6zaZGPh4Z7SXUiLtZaX1YIsSmGOdvF6Xb1WHI=";
+        } else if stdenvNoCC.isDarwin then {
+          url = "https://dl.google.com/android/maven2/com/android/tools/build/aapt2/${version}/aapt2-${version}-osx.jar";
+          hash = "sha256-hDfEPk3IJt+8FbRVEiHQbn24vsuOe6m36UcQsT6tGsQ=";
+        } else throw "Unsupport platform";
```
Comment (AndersonTorres) [inline pkgs/development/tools/aapt/default.nix:21]:
```suggestion
        } else throw "Unsupport platform: ${stdenv.system}";
```

File: pkgs/development/tools/aapt/default.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+    in
+    fetchzip (urlAndHash // {
+      extension = "zip";
+      stripRoot = false;
+    });
+
+  nativeBuildInputs = lib.optionals stdenvNoCC.isLinux [ autoPatchelfHook ];
+  buildInputs = lib.optionals stdenvNoCC.isLinux [ libcxx ];
+
+  installPhase = ''
```
Comment (AndersonTorres) [inline pkgs/development/tools/aapt/default.nix:31]: Don't forget `runHook`s

File: pkgs/development/tools/aapt/default.nix [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    install -D aapt2 $out/bin/aapt2
+  '';
+
+  meta = with lib; {
+    description = "A build tool that compiles and packages Android app's resources";
+    homepage = "https://developer.android.com/tools/aapt2";
+    sourceProvenance = with sourceTypes; [ binaryBytecode ];
```
Comment (AndersonTorres) [inline pkgs/development/tools/aapt/default.nix:38]:
```suggestion
```

File: pkgs/development/tools/aapt/default.nix [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    license = licenses.asl20;
+    maintainers = with maintainers; [ linsui ];
+    platforms = with platforms; unix;
+  };
```
Comment (AndersonTorres) [inline pkgs/development/tools/aapt/default.nix:42]:
```suggestion
    sourceProvenance = with lib.sourceTypes; [ binaryBytecode ];
  };
```

File: pkgs/development/tools/aapt/default.nix [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+    install -D aapt2 $out/bin/aapt2
+  '';
+
+  meta = with lib; {
+    description = "A build tool that compiles and packages Android app's resources";
+    homepage = "https://developer.android.com/tools/aapt2";
+    sourceProvenance = with sourceTypes; [ binaryBytecode ];
+    license = licenses.asl20;
+    maintainers = with maintainers; [ linsui ];
+    platforms = with platforms; unix;
```
Comment (AndersonTorres) [inline pkgs/development/tools/aapt/default.nix:41]:
```suggestion
    license = lib.licenses.asl20;
    maintainers = with lib.maintainers; [ linsui ];
    platforms = lib.platforms.unix;
```

File: pkgs/development/tools/aapt/default.nix [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    });
+
+  nativeBuildInputs = lib.optionals stdenvNoCC.isLinux [ autoPatchelfHook ];
+  buildInputs = lib.optionals stdenvNoCC.isLinux [ libcxx ];
+
+  installPhase = ''
+    install -D aapt2 $out/bin/aapt2
+  '';
+
+  meta = with lib; {
```
Comment (AndersonTorres) [inline pkgs/development/tools/aapt/default.nix:35]:
```suggestion
  meta = {
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

**Review by AndersonTorres — APPROVED** [review]

**Review by NickCao — APPROVED** [review]
Nice work!

**Pull request merged**